utils/CheckONNXModelZoo.py

```diff
@@ -0,0 +1,278 @@
+deprecated_models = {
+    "mnist-1",
+    "bvlcalexnet-3",
+    "caffenet-3",
+    "densenet-3",
+    "inception-v1-3",
+    "inception-v2-3",
+    "rcnn-ilsvrc13-3",
+    "resnet50-caffe2-v1-3",
+    "shufflenet-3",
+    "zfnet512-3",
+    "vgg19-caffe2-3",
+    "emotion-ferplus-2",
+}
+
+
+def obtain_all_model_paths():
+    _, model_paths = execute_commands(FIND_MODEL_PATHS_CMD)
+    model_paths = model_paths.split('\n')
+    # Remove empty paths and prune '._' in a path.
+    model_paths = [path[2:] for path in model_paths if path]
+    model_names = [
+        path.split('/')[-1][:-len(".tag.gz")] for path in model_paths
+    ]  # remove .tag.gz
+    deprecated_names = set(model_names).intersection(deprecated_models)
+
+    log_l1('\n')
+    deprecated_msg = ""
+    if (len(deprecated_names) != 0):
+        deprecated_msg = "where " + \
+            str(len(deprecated_names)) + \
+            " models are deprecated (using very old opsets, e.g. <= 3)"
+    log_l1("# There are {} models in the ONNX model zoo {}".format(
+        len(model_paths), deprecated_msg))
+    log_l1("See https://github.com/onnx/models/pull/389",
+           "for a list of deprecated models\n")
+    return model_names, model_paths
+
+
+def check_model(model_path, model_name, mcpu):
+    passed = False
+    with tempfile.TemporaryDirectory() as tmpdir:
+        # untar
+        log_l1('Extracting the .tag.gz to {}'.format(tmpdir))
+        execute_commands(UNTAR_CMD + [model_path, '-C', tmpdir])
+        _, onnx_files = execute_commands(
+            ['find', tmpdir, '-type', 'f', '-name', '*.onnx'])
+        # log_l1(onnx_files)
+        # temporary folder's structure:
+        #   - model.onnx
+        #   - test_data_set_0
+        #   - test_data_set_1
+        #   - test_data_set_2
+        #   - ...
+        #   - test_data_set_n
+
+        # Check .onnx file.
+        if (len(onnx_files) == 0):
+            log_l1("There is no .onnx file for this model. Quiting ...")
+            return passed
+        onnx_file = onnx_files.split('\n')[0]
+
+        # Check data sets.
+        has_data_sets = False
+        _, data_sets = execute_commands(
+            ['find', tmpdir, '-type', 'd', '-name', 'test_data_set*'])
+        if (len(data_sets) > 0):
+            has_data_sets = True
+            data_set = data_sets.split('\n')[0]
+        else:
+            # if there is no `test_data_set` subfolder, find a folder containing .pb files.
+            _, pb_files = execute_commands(
+                ['find', tmpdir, '-name', '*.pb', '-printf', '%h\n'])
+            if (len(pb_files) > 0):
+                has_data_sets = True
+                data_set = pb_files.split('\n')[0]
+        if (not has_data_sets):
+            log_l1("Warning: This model does not have test data sets.")
+
+        # compile, run and verify.
+        log_l1("Checking the model {} ...".format(model_name))
+        compile_args = '--compile_args=-O3'
```

Comment:
Would be nice to allow users of the script to set the onnx-mlir optimization level (and other compiler options) via the script command line interface.  By default, is the users does not overwrite the compiler options, I think is OK to compile at -O3.